### PR TITLE
Move non-opencontainer labeling of the image to breeze from Dockerfile

### LINKED ISSUE #21046: More ArtifactsHub specific labels in docker image
### Description

We added two labels that allow us to publish docker images in [ArtifactHub](https://artifacthub.io/). We should consider adding other labels as well, if it helps to exposing our image on ArtifactHub. https://artifacthub.io/docs/topics/repositories/#container-images-repositories

Probably, we should consider adding these labels with `docker build --label` i.e. as a docker command argument instead of Dockerfile instruction. For now, we should hold off on this change, as the image building process is undergoing a major overhaul - rewriting from Bash to Python.

### Use case/motivation

Better exposition of our images on ArtifactHub

### Related issues

https://github.com/apache/airflow/pull/21040#issuecomment-1019423007

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by potiuk**
Fixes: #21046

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
@mik-laj  - is that what you thought :) ?

File: dev/breeze/src/airflow_breeze/build_image/prod/build_prod_params.py [hunk 1]
```diff
@@ -202,6 +214,15 @@ def extra_docker_build_flags(self) -> List[str]:
                     f"AIRFLOW_CONSTRAINTS_REFERENCE={self.default_constraints_branch}",
                 ]
             )
+        extra_build_flags.extend(
+            [
+                "--label",
```
Comment (mik-laj) [inline dev/breeze/src/airflow_breeze/build_image/prod/build_prod_params.py:224]: What do you think aboutt adding other labels like `io.artifacthub.package.maintainers`, `io.artifacthub.package.logo-url`?

Comment (potiuk) [reply dev/breeze/src/airflow_breeze/build_image/prod/build_prod_params.py:224]: Sure. We can do it - care for adding them :) ?

Comment (mik-laj) [reply dev/breeze/src/airflow_breeze/build_image/prod/build_prod_params.py:224]: Will do.

Comment (mik-laj) [reply dev/breeze/src/airflow_breeze/build_image/prod/build_prod_params.py:224]: Done

**Review by mik-laj — APPROVED** [review]

**Pull request merged**